packages/tailwindcss/src/candidate.ts

```diff
@@ -777,4 +777,10 @@ function* findRoots(input: string, exists: (input: string) => boolean): Iterable
 
     idx = input.lastIndexOf('-', idx - 1)
   } while (idx > 0)
+
+  // Try '@' variant after permutations. This allows things like `@max` of `@max-foo-bar`
+  // to match before looking for `@`.
+  if (input[0] === '@' && exists('@')) {
+    yield ['@', input.slice(1)]
+  }
```

Comment:
I've been thinking of restructuring this somehow so that there's only one special-case for `@`. Let me know what you think about this (I pushed onto your PR). Tests seem to pass so it should be fine? 🤞